kiosk/src/main/index.js

```diff
@@ -1,62 +1,153 @@
-import { app, BrowserWindow, ipcMain } from 'electron'
+import { app, BrowserWindow, ipcMain } from "electron";
+const SerialPort = require("serialport");
+const Readline = require("@serialport/parser-readline");
 
 /**
  * Set `__static` path to static files in production
  * https://simulatedgreg.gitbooks.io/electron-vue/content/en/using-static-assets.html
  */
-if (process.env.NODE_ENV !== 'development') {
-  global.__static = require('path').join(__dirname, '/static').replace(/\\/g, '\\\\')
+if (process.env.NODE_ENV !== "development") {
+  global.__static = require("path")
+    .join(__dirname, "/static")
+    .replace(/\\/g, "\\\\");
 }
 
-let mainWindow
-const winURL = process.env.NODE_ENV === 'development'
-  ? `http://localhost:9080`
-  : `file://${__dirname}/index.html`
+let mainWindow;
+let parser;
+let oldData;
+let port;
+let scannerPort;
+
+const winURL =
+  process.env.NODE_ENV === "development"
+    ? `http://localhost:9080`
+    : `file://${__dirname}/index.html`;
 
 function createWindow() {
+  const shouldUseFrame = process.env.NODE_ENV === "development" ? true : false;
   /**
    * Initial window options
    */
   mainWindow = new BrowserWindow({
     height: 480,
     useContentSize: true,
-    width: 800
-  })
+    width: 800,
+    frame: shouldUseFrame
+  });
+
+  mainWindow.loadURL(winURL);
+
+  mainWindow.on("closed", () => {
+    mainWindow = null;
+  });
+
+  SerialPort.list((err, ports) => {
+    if (err) {
+      console.log("SerialPort.list err");
+      console.log(err);
+    }
 
-  mainWindow.loadURL(winURL)
+    console.log(ports);
 
-  mainWindow.on('closed', () => {
-    mainWindow = null
-  })
+    scannerPort = ports.find(port => {
+      return port.comName === "/dev/ttyUSB0";
```

Comment:
Will this scanner always be on this dev address?

Reply:
Or was the device id I was using not working with your HID?

Reply:
I can look into this more as using `/dev/ttyUSB0` is far from stable. I will try something out this weekend before I start work on the actual Kiosk

Reply:
👍